### PR TITLE
Adding `evotorch` from PyPI

### PR DESCRIPTION
**Pull request by sugatoray**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sugatoray** [comment]
## Reason for Build Failure: `1/N`

`Observation`:

> :point_right: *Use `ray-default` instead of `ray`*

- `line number: 864`: [(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=548995&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=864)
- `time elapsed`: `3m:22s`

```sh
The reported errors are:
- Encountered problems while solving:
-   - nothing provides requested ray >=1.0

```

**Comment by sugatoray** [comment]
## Reason for Build Failure: `?/N`

`Observation`:

> :point_right: *Need `opencensus-context>=0.1.3` to pass `pip check`.*

- `line number: 1689`: [(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=549000&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1689)
- `time elapsed`: `5m:18s`

```sh
import: 'evotorch'
+ pip check
opencensus 0.11.0 has requirement opencensus-context>=0.1.3, but you have opencensus-context 0.1.2.
```


<details>
<summary><b>Click to expand</b>: Detailed Error Trace</summary>
<p>

```sh
import: 'evotorch'
import: 'evotorch'
+ pip check
opencensus 0.11.0 has requirement opencensus-context>=0.1.3, but you have opencensus-context 0.1.2.
Tests failed for evotorch-0.1.1-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
WARNING:conda_build.build:Tests failed for evotorch-0.1.1-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
TESTS FAILED: evotorch-0.1.1-pyhd8ed1ab_0.tar.bz2
```

</p>
</details>

**Comment by sugatoray** [comment]
## Reason for Build Failure: `3/N`

`Observation`:

> :point_right: *The dependency `opencensus-context>=0.1.3` is not yet available on conda-forge. Raised a PR to update the package from `v0.1.2` to `v0.1.3`.*

- `line number: 867`: [(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=549019&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=867)
- `time elapsed`: `3m:49s`

```sh
The reported errors are:
- Encountered problems while solving:
-   - nothing provides requested opencensus-context >=0.1.3
```

---

PR raised to update `opencensus-context` to version `0.1.3`.

- https://github.com/conda-forge/opencensus-context-feedstock/pull/15

**Comment by sugatoray** [comment]
## Note

Windows build fails as `pytorch` is not available for windows on `conda-forge`.

**Comment by sugatoray** [comment]
PR is ready. Request for review/merging.

cc: @conda-forge/help-python

**Comment by sugatoray** [comment]
Hi @BastianZim

I am planning to showcase a few recipes (already existing ones on conda-forge) in the proposed conda-forge workshop. 

What do you think of this one? 

The fact that you don't have PyTorch available for all platforms and windows build fails because of that and similarly OSX fails because of `ray-default`, will help set some expectations. 

Also, that PyPI `ray` is `ray-default ` on conda-forge and that one should check the package names on conda-forge.org, will help the workshop attendees debug their recipes.

**Comment by BastianZim** [comment]
Hi @sugatoray 

yes, I think that's a great idea. These edge cases are what really is important as most are nowadays solved with Grayskull.

> Also, that PyPI ray is ray-default on conda-forge and that one should check the package names on conda-forge.org, which will help the workshop attendees debug their recipes.

Maybe also point out that Grayskull maintains a mapping for that which should be updated if something like this is added: https://github.com/conda-incubator/grayskull/blob/main/grayskull/strategy/config.yaml

Re the review:

> Resolution: Disable pip check

Shouldn't the package be added to conda-forge instead?

**Comment by BastianZim** [comment]
Ahh ok, but are you using different versions here than what upstream recommends? Are you sure this can be loosened?

**Comment by sugatoray** [comment]
> Maybe also point out that Grayskull maintains a mapping for that which should be updated if something like this is added: https://github.com/conda-incubator/grayskull/blob/main/grayskull/strategy/config.yaml

Yes. This is on the cards. Also there is another list that is maintained for the pypi---conda-forge mapping. It's rather exhaustive. The grayskull utility list does not cover all of them.

**Comment by sugatoray** [comment]
> Ahh ok, but are you using different versions here than what upstream recommends? Are you sure this can be loosened?

Upstream does not include this (`opencensus-context`) as a dependency at all. It only comes up during pip check.

**Comment by BastianZim** [comment]
Hmm weird ok. I am away form my computer today but will merge this asap.

File: recipes/evotorch/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/evotorch-{{ version }}.tar.gz
+  sha256: bd1f612ef8709e04d59e197aca618bbe579229b13af205d64ecc671535893d57
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
```
Comment (BastianZim) [inline recipes/evotorch/meta.yaml:18]: Might be good to include those? https://github.com/nnaisense/evotorch/blob/2a6ad9404ff6a1ec625f53d116ea7ed6adab0991/pyproject.toml#L31

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just one small thing

**Comment by sugatoray** [comment]
@BastinZim My update for opencensus-context feedstock was merged. So, I believe the pip check will work now. 

Also, I will add the build tools (from pyproject.toml) to host-requirements (as you suggested). 

Let me make these changes first before you consider merging.

**Pull request closed without merging**